packages/backend-common/src/logging/logger.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import winston, { Logger } from 'winston';
+
+export let logger: Logger = winston.createLogger({
+  level: process.env.LOG_LEVEL || 'info',
+  format:
+    process.env.NODE_ENV === 'production'
+      ? winston.format.json()
+      : winston.format.combine(
+          winston.format.colorize(),
+          winston.format.timestamp(),
+          winston.format.simple(),
+        ),
+  defaultMeta: { service: 'backstage' },
+  transports: [
+    new winston.transports.Console({
+      silent:
+        process.env.JEST_WORKER_ID !== undefined && !process.env.LOG_LEVEL,
+    }),
+  ],
+});
+
+export function setLogger(newLogger: Logger) {
```

Comment:
Maybe. How would we scope and create them?

So, maybe we could supply a `createPlugin` function in the backend too, that returns a Router / sub-app. It would also create a logger with the necessary properties, and do `subapp.set('logger', logger)` behind the scenes. Then there could be a framework api like  `log.info(req: Request, message: string)` that essentially does `req.app.get('logger').info(...)` or so. Pseudo code, gotta look at the apis :)

Reply:
Yeah also been thinking about mirroring the backend APIs to the frontend APIs a bit, just for familiarity where it makes sense. We should settle the frontend APIs a bit more first though and just go with whatever makes sense for the backend imo x)

When it comes to accessing the logger we could go with a pattern of something like plugins exporting creation functions that will be called with a fixed set of parameters + extra config like discussed in the scaffolder PR.

The actual creation of the logger instance itself would be `logger.child()`